state/migrations/20230802121023_device_data_jsonb.go

```diff
@@ -33,10 +34,32 @@ func upJSONB(ctx context.Context, tx *sql.Tx) error {
 	if err != nil {
 		return err
 	}
-	_, err = tx.ExecContext(ctx, "UPDATE syncv3_device_data SET dataj = encode(data, 'escape')::JSONB;")
+
+	rows, err := tx.Query("SELECT user_id, device_id, data FROM syncv3_device_data")
 	if err != nil {
 		return err
 	}
+
+	// abusing PollerID here
+	var deviceData sync2.PollerID
+	var data []byte
+
+	// map from PollerID -> deviceData
+	deviceDatas := make(map[sync2.PollerID][]byte)
+	for rows.Next() {
+		if err = rows.Scan(&deviceData.UserID, &deviceData.DeviceID, &data); err != nil {
+			return err
+		}
+		deviceDatas[deviceData] = data
+	}
+
+	for dd, d := range deviceDatas {
+		_, err = tx.ExecContext(ctx, "UPDATE syncv3_device_data SET dataj = $1 WHERE user_id = $2 AND device_id = $3;", d, dd.UserID, dd.DeviceID)
+		if err != nil {
+			return err
```

Comment:
How does this help when there are problems casting to jsonb?

Reply:
It's not the casting to jsonb, it's the `encode(data, 'escape')` which adds e.g. `\3` which then can't be casted to jsonb